plugins/api/src/main/java/org/opennms/poc/plugin/api/ParameterMap.java

```diff
@@ -0,0 +1,334 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2002-2014 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2014 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.poc.plugin.api;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Convenience class for looking up string and integer values in a parameter
+ * map.
+ */
+public abstract class ParameterMap {
```

Comment:
A generic idea, as I have no other points so far - why not making parameters a type on its own.

I see that quite often validation of various configuration parameters becomes a burden, especially if some of them are optional or conditional. If we define a `Parameter` and `Validator` interfaces we gain a full flexibility in how plugins of various kinds can be configured. Below utility can still remain, but not as a way to handle configuration injection.

Further design improvement if above is insufficient could be separation of parameter sourcing. We then leave on plugin end a input schema (parameter definitions) with validation rules and unify injection/configuration logic - ie. file, database or workflow definition.

Reply:
A cool idea, but personally I favor blueprint placeholder .cfg files for configuring plugins. Makes it simple to inject the values into the beans, and still gives us better type safety. 

But that I suspect is debate for the main horizon project, I just pulled this in to ge the POC working easy.